### DISCUSSION TITLE
aspnet:7.0 - unable to start container process: exec: "dotnet": executable file not found in $PATH

### DISCUSSION THREAD

**Discussion by yuvron** [dotnet/dotnet-docker#4982] [Q&A]
### Describe the Bug

Can't run any `mcr.microsoft.com/dotnet/aspnet:7.0` based image.
My Dockerfile:
```Dockerfile
FROM mcr.microsoft.com/dotnet/aspnet:7.0
WORKDIR /app
COPY . .
CMD ["dotnet", "main.dll"]
```
When I run this image I receive this error:
`docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "dotnet": executable file not found in $PATH: unknown.`

Switching to other tag like `7.0-bookworm-slim` solved the problem.
### Other Information

```
uname -a
Linux tegra-ubuntu 4.9.253-tegra #1 SMP PREEMPT Tue Jan 3 15:26:52 IST 2023 aarch64 aarch64 aarch64 GNU/Linux
```

### Output of `docker version`

```
Docker version 24.0.2, build cb74dfc
```

### Output of `docker info`

```
Client: Docker Engine - Community
 Version:    24.0.2
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.10.5
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.18.1
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 11
  Running: 7
  Paused: 0
  Stopped: 4
 Images: 20
 Server Version: 24.0.2
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 nvidia runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3dce8eb055cbb6872793272b4f20ed16117344f8
 runc version: v1.1.7-0-g860f061
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
 Kernel Version: 4.9.253-tegra
 Operating System: Ubuntu 18.04.5 LTS
 OSType: linux
 Architecture: aarch64
 CPUs: 6
 Total Memory: 15.45GiB
 Name: tegra-ubuntu
 ID: fcc49e5e-3e1d-4135-966a-40726bf3e2e4
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Comment by lbussell** [comment]
@yuvron, Can you share the digest (sha256) of the `mcr.microsoft.com/dotnet/aspnet:7.0` base image you're using The latest that I see for arm64 (for your Tegra machine) is `4e1e37fee2df5877fc3cd2723b0a77c0887a496b2bcce0303693c3b0ec35ab0d`. It should have the `dotnet` command defined:

```bash
> docker run --rm mcr.microsoft.com/dotnet/aspnet:7.0-bullseye-slim-arm64v8 which dotnet
/usr/bin/dotnet
```

If the base image isn't up-to-date, please try adding the `--pull` argument to your image build. If that doesn't work, maybe you could try using `ENTRYPOINT` instead of `CMD` as we do in our samples:

https://github.com/dotnet/dotnet-docker/blob/1a88de08b1fef937c628b877d8300634f8f650fe/samples/aspnetapp/Dockerfile#L21

Either `ENTRYPOINT [ "dotnet", "aspnetapp.dll" ]` or `ENTRYPOINT [ "./aspnetapp" ]` should work as they are both output by `dotnet publish` from our .NET 7 samples, unless you changed the publish command in your Dockerfile.

**Comment by lbussell** [comment]
[Triage] Moving this to a discussion as it's not clear yet that this is a bug.

**Comment by rowanfuchs** [comment]
> @yuvron, 
> Either `ENTRYPOINT [ "dotnet", "aspnetapp.dll" ]` or `ENTRYPOINT [ "./aspnetapp" ]` should work as they are both output by `dotnet publish` from our .NET 7 samples, unless you changed the publish command in your Dockerfile.

If this isn't working, I assume you are using Rider, and you will need to configure Docker again, as this was the case for me. With a new install, mine defaulted to "TCP Socket" while I am using WSL. After setting it to WSL and the chosen distro, it ran as expected.
